test/gtest/uct/test_md.cc

```diff
@@ -358,6 +376,49 @@ UCS_TEST_P(test_md, reg_multi_thread) {
     pthread_join(thread_id, NULL);
 }
 
+UCS_TEST_P(test_md, sockaddr_accessibility) {
+    ucs_sock_addr_t sock_addr;
+    struct ifaddrs *ifaddr, *ifa;
+    struct ifreq if_req;
+    ucs_status_t status;
+    int rc, found_ipoib = 0;
+
+    check_caps(UCT_MD_FLAG_SOCKADDR, "sockaddr");
+
+    rc = getifaddrs(&ifaddr);
+    ASSERT_TRUE(rc != -1);
+
+   /* go through a linked list of available interfaces and try to find an
+    * IPoIB interface */
+    for (ifa = ifaddr; ifa != NULL; ifa = ifa->ifa_next) {
+         status = ucs_netif_ioctl(ifa->ifa_name, SIOCGIFHWADDR, &if_req);
+         ASSERT_UCS_OK(status);
+         /* look for an Infiniband interface */
+         if (if_req.ifr_addr.sa_family == ARPHRD_INFINIBAND) {
+             /* try to find an IPv4 address on the interface */
+             if (ifa->ifa_addr->sa_family == AF_INET) {
+                 found_ipoib = 1;
+                 break;
+             }
+         }
+    }
+
+    if (found_ipoib) {
```

Comment:
this test is generic and should not assume/test ipoib.
need to either not fail the test if the address is not accessible, or move this test to test/uct/ib

Reply:
what do you mean? if there is no ipoib address, the test is skipped. if the address is there but not accessible with rdmacm, the test will fail.
rdmacm must have an ipoib address to work.

Reply:
but this test is not only for rdmacm, it's for uct sockaddr connection establishment. it should work with other md's which would support sockaddr, e.g tcp

Reply:
tcp won't work with this ip?

Reply:
it will, but tcp should be tested even if ipoib is not present